### PR TITLE
to_hf.py: add --vllm flag for vLLM-ready checkpoint export

### PR DESCRIPTION
**Pull request by DongjiGao**
## Summary

- Add `vllm=True` flag to `to_hf.py` that produces vLLM-ready checkpoints in one step
- `prepare_for_vllm()` runs after checkpoint consolidation (rank 0 only) and handles architecture detection, chat template generation, tokenizer saving, and generation config
- Depends on #15616 (prompt formatters `to_jinja()`)

### `examples/speechlm2/to_hf.py`
- `HfExportConfig.vllm: bool = False` — opt-in flag
- `_detect_vllm_architecture()` — determines hybrid vs standard from `pretrained_llm`
- `prepare_for_vllm(output_dir, model_cfg)`:
  1. Patches `config.json` with `model_type: "nemo_speechlm"` and detected `architectures`
  2. Calls `PromptFormatter.resolve(prompt_format).to_jinja()` for chat template
  3. Saves tokenizer with `<|audio|>` token and chat template
  4. Writes minimal `generation_config.json` (EOS only — inference params are task-specific)
- Raises `ValueError` for missing `prompt_format`/`pretrained_llm`
- Warns when overwriting existing files

### Architecture mapping
| LLM backbone | Architecture name |
| ------------ | ----------------- |
| Hybrid (NemotronH) | `NeMoSpeechLMHybridForConditionalGeneration` |
| Standard (Qwen3, etc.) | `NeMoSpeechLMForConditionalGeneration` |

## Usage
```bash
torchrun --nproc-per-node=8 --nnodes=4 \
    examples/speechlm2/to_hf.py \
    class_path=nemo.collections.speechlm2.models.salm_automodel.SALMAutomodel \
    ckpt_path=/path/to/checkpoint.ckpt \
    ckpt_config=/path/to/exp_config.yaml \
    output_dir=/path/to/hf_output \
    vllm=True
```

## Test plan
- [x] End-to-end: `to_hf.py vllm=True` on nano-v3-canary1bflash (32 GPUs, 5.5 min)
- [x] Output verified: correct `model_type`, `architectures`, chat template (1149 chars), `generation_config.json`
- [x] Eval: `burst_eval_vllm.py` on produced checkpoint — WER matches baseline (6.31 overall)
- [ ] CI: add "Run CICD" label

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: examples/speechlm2/to_hf.py [hunk 1]
```diff
@@ -102,10 +106,101 @@ def save_hf_checkpoint(model: torch.nn.Module, state_dict: dict, cfg: HfExportCo
...
+
+def _detect_vllm_architecture(model_cfg: dict) -> str:
+    """Determine the vLLM plugin model class from the pretrained LLM backbone."""
+    pretrained_llm = model_cfg.get("pretrained_llm", "")
+    try:
+        from transformers import AutoConfig
+
+        llm_cfg = AutoConfig.from_pretrained(pretrained_llm, trust_remote_code=True)
+        archs = getattr(llm_cfg, "architectures", [])
+    except Exception:
```
Comment (pzelasko) [inline examples/speechlm2/to_hf.py:128]: Broad exception clause, what errors are you expecting here?

File: examples/speechlm2/to_hf.py [hunk 2]
```diff
@@ -42,6 +42,10 @@ class HfExportConfig:
     # Dtype used for stored parameters
     dtype: str = "bfloat16"
 
+    # If True, patch the checkpoint for vLLM inference (add tokenizer, chat template,
+    # model_type/architectures, generation_config). Requires HuggingFace transformers.
+    vllm: bool = False
```
Comment (pzelasko) [inline examples/speechlm2/to_hf.py:47]: Remove this flag and always output converted checkpoints that work with both HF and vLLM engines
